### ISSUE TITLE
[BUG] Cannot Provision RKE1 on vSphere

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#47938]
**Rancher Server Setup**
- Rancher version: v2.9.3 
- Installation option (Docker install/Helm Chart): any 
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version:  any 
- Cluster Type (Local/Downstream):
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Downstream, RKE1 node-driver vSphere 

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) Admin 
 

**Describe the bug**
 
Currently, attempting to provision a rke1 cluster on vsphere results in the following error in both Ubuntu and Rocky (tested so far):

```
Error setting machine configuration from flags provided: open /FAM2EQU42T/cloud-config: permission denied:Timeout waiting for ssh key 
```

This occurs when creating new clusters or adding nodes to an existing cluster if Rancher has been upgraded from an older version.

**To Reproduce**

- Spin up Rancher v2.9.3
- Add vSphere Cloud Creds
- Create RKE1 Node Template
- Create new RKE1 cluster w/vSphere node driver
- **No extra options**, also tried multiple K8S versions  

**Result**

The following error in Rancher UI on the nodes
```
Error setting machine configuration from flags provided: open /FAM2EQU42T/cloud-config: permission denied:Timeout waiting for ssh key 
```

<img width="700" alt="screenshot-2" src="https://github.com/user-attachments/assets/993df777-2eeb-45c3-8d18-a73da0d1ebab">

**Expected Result**

Cluster provision should succeed 
 

**Additional Inforamtion** 

When creating a Node Template for vSphere through the Rancher UI, the UI appends the string `#cloud-config` to the `Cloud Config` field by default. See the following screenshot

<img width="616" alt="screenshot-1 (1)" src="https://github.com/user-attachments/assets/42d40621-b0d8-49fb-81a1-169d4133263a">

SURE-9329

**Comment by jiaqiluo** [comment]
The fix can be validated on the latest v2.10-head image

**Comment by susesgartner** [comment]
Ticket #47938 - Test Results - ✅ 
---
 
| Scenario | Test Case | Result |
|---|---|---|
| 1. | provision RKE1 cluster | PASS |
 
---
Verified with (HA Helm or Docker) on Rancher `v2.10-head`:
---
 
Scenario 1: provision RKE1 cluster
---
1. Provision an RKE1 cluster 
2. Verify that no permission error is observed
3. Login to one of the nodes
4. Verify that the permissions for cloud-config are set properly
 
Results: No error was observed the cluster provisioned successfully
